### PR TITLE
KCL stdlib 'map' function

### PR DESCRIPTION
**Pull request by adamchalmers**
Takes an array `[a, b, ...]` and a function `f`, returns `[f(a), f(b), ...]`

### CODE REVIEW

**Review by jtran — COMMENTED** [review]
Looks good so far.

File: src/wasm-lib/kcl/src/std/args.rs [hunk 1]
```diff
@@ -484,6 +485,7 @@ where
             }));
         };
         let Some(val) = T::from_mem_item(arg) else {
+            eprintln!("Adam1");
```
Comment (jtran) [inline src/wasm-lib/kcl/src/std/args.rs:488]:
```suggestion
```

**Review by jtran — APPROVED** [review]
Cool!

File: src/wasm-lib/kcl/src/std/args.rs (hunk 1 above)
Comment (adamchalmers) [reply src/wasm-lib/kcl/src/std/args.rs]: Thanks, fixed.

**Pull request merged**